pkg/manager/impl/execution_manager.go

```diff
@@ -184,12 +182,49 @@ func (m *ExecutionManager) addPluginOverrides(ctx context.Context, executionID *
 	return nil, nil
 }
 
-// TODO: Delete this code usage after the flyte v0.17.0 release
+// defaults should be a coalesce of task defaults, and platform defaults.
+// task limits should be the limits from the task, coalesced with the defaults from step one
+// then both should be limited by any platform limits.
+// anything 0 or empty is not set.
+// if both requests and limits end up empty, return nil. if one is empty, return nil for it
+func (m *ExecutionManager) getResources(ctx context.Context, taskResources *core.Resources, platformResources workflowengineInterfaces.TaskResources) *core.Resources {
+
+	// requests: coalesce(task request, platform default)
+	// limits: coalesce(task limits, task requests)
```

Comment:
IMHO this only makes sense for non-compressible resources like memory and storage.
Compressible resources like cpu should not be "forced" to inherit limits from requests if not explicitly specified. At the very least the limit should be explicitly removable/nullable.